src/test/java/com/google/sps/UserTripServletTest.java

```diff
@@ -0,0 +1,167 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.Config;
+import com.google.sps.data.User;
+import com.google.sps.servlets.AuthServlet;
+import com.google.sps.servlets.UserTripServlet;
+import com.google.sps.servlets.TripServlet;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(UserServiceFactory.class)
+public final class UserTripServletTest {
+
+  // Add constants necessary to testing retrieval of the current User.
+  private static final String EMAIL = "testemail@gmail.com";
+  private static final String AUTH_DOMAIN = "gmail.com";
+  private static final String LOGOUT_URL = "/_ah/logout?continue=%2F";
+  private static final String LOGIN_URL = "/_ah/login?continue=%2F";
+
+  // Create UserTripServlet and TripServlet objects.
+  private UserTripServlet userTripServlet;
+  private TripServlet tripServlet;
+
+  // Add helper to allow datastore testing in local JUnit tests.
+  // See https://cloud.google.com/appengine/docs/standard/java/tools/localunittesting.
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void initServlets() {
+    userTripServlet = new UserTripServlet();
+    tripServlet = new TripServlet();
+  }
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testDoGetResponseLoggedIn() throws Exception {
```

Comment:
I feel like there are some test cases you haven't covered here e.g. no trips returned from datastore and multiple trips returned from datastore

Reply:
That's true, I'll be sure to add these tests.

Reply:
I added these, and removed the tests that check whether or not the user is logged in, as these aren't necessary anymore as a result of my new `UserTripServlet.java` file structure.